sensor_topic:
  image_topic: /kitti/camera_color_right/image_raw
  imu_topic: /kitti/oxts/imu

feature_extractor:
    confidence_thresh: 0.015
    distance_thresh: 8
    horizontal_boarder_width: 10
    vertical_boarder_width: 10
    inference_width: 1280
    inference_height: 320
    weight_path: /home/chenghe/catkin_ws/src/SuperVIO/data/superpoint.pt

feature_tracker:
  match_thresh: 0.6
  parallax_thresh: 10.0
  ratio_thresh: 0.6
  fundamental_filter: true

triangulator:
  view_angle_check: false
  baseline_check: false
  baseline_thresh: 0.1
  error_thresh: 1.0
  view_angle_thresh: 0.1

optimizer:
  linear_solver_type: "dense_schur"
  trust_region_strategy: "dogleg"
  max_iteration: 50
  num_threads: 8
  verbose: false
  max_solver_time: 1.0

estimation:
  min_num_match: 20
  sliding_window_size: 5
  parallax_thresh: 10.0
  linearized_ba_thresh: 0.1
  linearized_bg_thresh: 0.1
  max_preintegration_time: 10.0

imu:
  accelerator_noise: 0.1
  gyroscope_noise: 0.1
  accelerator_random_walk_noise: 0.001
  gyroscope_random_walk_noise: 0.001

camera:
  camera_type: pinhole
  input_intrinsic:  [ 903.7596, 901.9653, 695.7519,  224.2509]
  input_distortion: [-0.3639558, 0.1788651, 0.0006029694, -0.0003922424, -0.0538246]
  output_intrinsic: [ 903.7596, 901.9653, 695.7519,  224.2509]
  input_image_height: 375
  input_image_width: 1242
  output_image_height: 375
  output_image_width: 1242
  use_gpu_for_undistortion: true

extrinsic:
  extrinsic_rotation: [0.501288465177, -0.499208635594, 0.496572701836, -0.502907683873]
  extrinsic_position: [1.0837718794, -0.780440771155, 0.731924112461]
  gravity_norm: 9.81007
